versions/3.0.4.md

```diff
@@ -3495,3 +3497,53 @@ Version   | Date       | Notes
 1.2       | 2014-03-14 | Initial release of the formal document.
 1.1       | 2012-08-22 | Release of Swagger 1.1
 1.0       | 2011-08-10 | First release of the Swagger Specification
+
+## <a name="usingRFC6570Implementations"></a>Appendix C: Using RFC6570 Implementations
+
+Serialization is defined in terms of RFC6570 URI Templates in two scenarios:
+
+Object | Condition
+------ | ---------
+[Parameter Object](#parameterObject) | When `schema` is present
+[Encoding Object](#encodingObject) | When encoding for `application/x-www-form-urlencoded` and any of `style`, `explode`, or `allowReserved` are used
+
+Implementations of this specification MAY use an implementation of RFC6570 to perform variable expansion, however, some caveats apply.
+
+### Equivalences Between Fields and RFC6570 Operators
+
+Certain field values translate to RFC6570 operators (or lack thereof):
+
+field | value | operator
+----- | ----- | --------
+style | simple | _n/a_
+style | matrix | `;`
+style | label | `.`
+style | form | `?`
+allowReserved | `false` | _n/a_
+allowReserved | `true` | `+`
+explode | `false` | _n/a_
+explode | `true` | `*`
+
+Multiple `style: form` parameters are equivalent to a single RFC6570 [variable list](https://www.rfc-editor.org/rfc/rfc6570#section-2.2) using the `?` operator:
+
+```YAML
+- name: foo
+  in: query
+  schema:
+    type: object
+  explode: true
+- name: bar
+  in: query
+  schema:
+    type: string
+```
+
+This example is equivalent to RFC6570's `{?foo*,bar}`, and ***NOT*** `{?foo*}{&bar}`.
```

Comment:
> {?foo*}{&bar}
Is this even a valid URI Template?  If foo is empty/undefined then the resulting href would be invalid.

Reply:
Yeah that's kind of my point.  I mean, it's a valid URI Tempate in the sense that it's syntactically legal (I think?) it just might not produce a valid URI.  The URI Template spec is pretty intentionally blind to the context outside of the variables.

But perhaps I could note that that's why you need to do it the `{?foo*,bar}` way.

Reply:
I think I've addressed this in the latest, much-expanded version.